### PR TITLE
web: add filter to own commits button

### PR DESCRIPTION
**Pull request by scallaway**
Ultimately, this PR does what it says on the tin, but I included two other refactoring-ish commits ahead of them with other stuff that I found whilst working on this.

### CODE REVIEW

File: qvet-web/src/components/CommitSummary.tsx [hunk 1]
```diff
@@ -178,24 +181,41 @@ const CommitFiltering = ({
...
-            size="small"
-            sx={{
-              visibility: search ? "visible" : "hidden",
-            }}>
-            <ClearIcon />
-          </IconButton>
-        ),
-      }}
-    />
+    <Stack spacing={1} alignItems="flex-end">
```
Comment (tommilligan) [inline qvet-web/src/components/CommitSummary.tsx:187]: This `Stack` doesn't look like it does anything - it only contains one element, and has no padding?

File: qvet-web/src/components/CommitSummary.tsx [hunk 2]
```diff
@@ -178,24 +181,41 @@ const CommitFiltering = ({
...
+                onClick={() => setSearch("")}
+                size="small"
+                sx={{
+                  visibility: search ? "visible" : "hidden",
+                }}>
+                <ClearIcon />
+              </IconButton>
+              {loginData.data ? (
+                <IconButton
+                  onClick={() => setSearch(loginData.data.login)}
```
Comment (tommilligan) [inline qvet-web/src/components/CommitSummary.tsx:206]: Should this (and `onChange` above) be `useCallback`'d to avoid child rerendering?

**Review by tommilligan — APPROVED** [review]
Looks great, just two code points but happy

File: qvet-web/src/components/CommitSummary.tsx (hunk 1 above)
Comment (scallaway) [reply qvet-web/src/components/CommitSummary.tsx:187]: Ah whoops - mistake when building the commits back up

**Review by tommilligan — APPROVED** [review]
LG, thanks!

**Pull request merged**